hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/HasCallAttempt.java

```diff
@@ -0,0 +1,144 @@
+            @NonNull final List<CallTranslator> callTranslators,
+            final boolean isStaticCall) {
+        super(
+                input,
+                senderAddress,
+                authorizingAddress,
+                onlyDelegatableContractKeysActive,
+                enhancement,
+                configuration,
+                addressIdConverter,
+                verificationStrategies,
+                gasCalculator,
+                callTranslators,
+                isStaticCall,
+                REDIRECT_FOR_ACCOUNT);
+
+        if (this.isRedirect) {
+            this.redirectAccount = linkedAccount(redirectAddress);
+        } else {
+            redirectAccount = null;
+        }
+    }
+
+    /**
+     * Returns whether this is a account redirect.
+     *
+     * @return whether this is a account redirect
+     * @throws IllegalStateException if this is not a valid call
+     */
+    public boolean isAccountRedirect() {
+        return isRedirect;
+    }
+
+    /**
+     * Returns the account that is the target of this redirect, if it existed.
+     *
+     * @return the account that is the target of this redirect, or null if it didn't exist
+     * @throws IllegalStateException if this is not an account redirect
+     */
+    public @Nullable Account redirectAccount() {
+        if (!isRedirect) {
+            throw new IllegalStateException("Not an account redirect");
+        }
+        return redirectAccount;
+    }
+
+    /**
+     * Returns the id of the account that is the target of this redirect, if it existed.
+     *
+     * @return the id of the account that is the target of this redirect, or null if it didn't exist
+     * @throws IllegalStateException if this is not an account redirect
+     */
+    public @Nullable AccountID redirectAccountId() {
+        if (!isRedirect) {
+            throw new IllegalStateException("Not a account redirect");
+        }
+        return redirectAccount == null ? null : redirectAccount.accountId();
+    }
+
+    /**
+     * Returns the account at the given Besu address, if it exists.
+     *
+     * @param accountAddress the Besu address of the account to look up
+     * @return the account that is the target of this redirect, or null if it didn't exist
+     */
+    public @Nullable Account linkedAccount(@NonNull final Address accountAddress) {
+        requireNonNull(accountAddress);
+        return linkedAccount(accountAddress.toArray());
+    }
+
+    /**
+     * Returns the account at the given EVM address, if it exists.
+     *
+     * @param evmAddress the headlong address of the account to look up
+     * @return the account that is the target of this redirect, or null if it didn't exist
+     */
+    public @Nullable Account linkedAccount(@NonNull final byte[] evmAddress) {
+        requireNonNull(evmAddress);
+        return enhancement.nativeOperations().getAccount(numberOfLongZero(evmAddress));
```

Comment:
⚠️ Can we really assume that `evmAddress` is a long-zero address?

Reply:
No.  I'll add code to handle long-zero or evm addresses.

Reply:
ended up looking up the account from the store.  Is there a performance concern?